framework/src/Volo.Abp.MultiTenancy/Volo/Abp/MultiTenancy/CurrentImpersonatorTenant.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Security.Principal;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.Security.Claims;
+
+namespace Volo.Abp.MultiTenancy
+{
+    public class CurrentImpersonatorTenant : ICurrentImpersonatorTenant, ITransientDependency
```

Comment:
Instead of introducing `ICurrentImpersonatorTenant`, just add a `ImpersonatorId` to the `ICurrentTenant`.